### ISSUE TITLE
Deleting any namespace stuck with "unable to retrieve the complete list of server APIs: allocation.agones.dev/v1"

### ISSUE THREAD

**Issue by guitarrapc** [googleforgames/agones#2903]
**What happened**:

When installing Agones to the Cluster, deleting any namespace stucked with `Terminating` status happens occasionally.

**What you expected to happen**:

Namespace can delete without stucked via Agones.

**How to reproduce it (as minimally and precisely as possible)**:

I can't reproduce issue in minimum step. However it happens once a month since Agones 1.19 since end of 2021.

Using following Helm installation command.
```
helm upgrade --install agones agones/agones --version 1.24.0 --namespace agones-system --create-namespace --set agones.ping.install=false --set agones.allocator.replicas=1
```

**Anything else we need to know?**:

You can observe `kubectl describe ns SOME_NS_SUCKED_ON_TERMINATING` or `kubectl api-resources` reports `[unable to retrieve the complete list of server APIs: allocation.agones.dev/v1](error: unable to retrieve the complete list of server APIs: allocation.agones.dev/v1: the server is currently unable to handle the request)` .

**Environment**:
- Agones version: 1.24.0
- Kubernetes version (use `kubectl version`): Server Version: v1.24.7-eks-fb459a0
- Cloud provider or hardware configuration: AWS EKS
- Install method (yaml/helm): helm
- Troubleshooting guide log(s):

Sample condition of kubectl describe ns.

```
Name:         foo
Status:       Terminating
Conditions:
  Type                                         Status  LastTransitionTime               Reason                  Message
  ----                                         ------  ------------------               ------                  -------
  NamespaceDeletionDiscoveryFailure            True    Fri, 13 Jan 2023 12:04:32 +0900  DiscoveryFailed         Discovery failed for some groups, 1 failing: unable to retrieve the complete list of server APIs: allocation.agones.dev/v1: the server is currently unable to handle the request
  NamespaceDeletionGroupVersionParsingFailure  False   Fri, 13 Jan 2023 12:04:34 +0900  ParsedGroupVersions     All legacy kube types successfully parsed        
  NamespaceDeletionContentFailure              False   Fri, 13 Jan 2023 12:04:34 +0900  ContentDeleted          All content successfully deleted, may be waiting on finalization
  NamespaceContentRemaining                    False   Fri, 13 Jan 2023 12:05:07 +0900  ContentRemoved          All content successfully removed
  NamespaceFinalizersRemaining                 False   Fri, 13 Jan 2023 12:05:07 +0900  ContentHasNoFinalizers  All content-preserving finalizers finished       
```

- Others:

  * diff of `kubectl api-resources`, left = when everything working fine, right = when issue happen.
![image](https://user-images.githubusercontent.com/3856350/212233137-b70c0811-62f1-4fe3-adca-73a15e892dea.png)

  * https://github.com/googleforgames/agones/issues/1778 is when deleting agones-system, but this issue is when deleting **any** namespace. So I think this is difference issue.

**Workaround**

Reinstall agones helm fix issue.

step1. Remove all Agones related resources, like fleet, fleetautoscaler, gameserver and so on.
step2. Uninstall agones helm.
step3. Reinstall agones helm, then issue goaway.

```
helm uninstall -n agones-system agones
helm install -n agones-system agones
```

Unfortunately, recareating agones related pod won't fix issue.

**Comment by guitarrapc** [comment]
> Is the Agones controller running when you attempt this?

yes. 

Because controller is up I have no idea so reported this issue. Any idea is appreciated.

**Comment by markmandel** [comment]
🤔 two thoughts:

https://agones.dev/site/docs/guides/troubleshooting/#how-do-i-see-the-logs-for-agones

Anything in the Agones logs?

The other thought is that restarting the Pod doesn't solve the issue, but reinstalling the Helm chart does - could something be deleting either the Service for the API service, or even the record for the APIService itself within the cluster?

Some kind of policy manager perhaps? I'm not really familiar with EKS in the front

**Comment by guitarrapc** [comment]
Thanks. I have 3 clusters installed Agones, but only one of it replicated issue once a month but others are working fine. Only the difference is issued cluster has over 50 deployment a day. So I doubt some potential busy-ness related issue.

> Anything in the Agones logs?

I will check Agones logs when issue reproduced.

>  could something be deleting either the Service for the API service, or ...

I think no, but I agress deleting APIService is suspecious.

**Comment by zmerlynn** [comment]
Agree with @markmandel here - also check the uptime on the `agones-controller` pod, it may have been bounced.

I saw this failure mode in a past life: Extension APIs can block namespace deletion because `kube-apiserver` needs the discovery doc for any registered extensions so that it knows what resource types to tear down in the namespace (AIUI). You can typically force namespace deletion past this point, but it may leave dangling resources.

**Comment by markmandel** [comment]
I think we should probbaly move this to `wontfix` and close it. It's not something I think we can actually fix?

**Comment by guitarrapc** [comment]
Thanks. I agree I'm not confident Agones need anything fix, yet. Let's close and will report if found definite clue with.
